exec/java-exec/src/main/java/org/apache/drill/exec/store/sys/ExtendedOptionIterator.java

```diff
@@ -0,0 +1,152 @@
+
+  private final OptionManager fragmentOptions;
+  private final Iterator<OptionValue> mergedOptions;
+
+  public ExtendedOptionIterator(FragmentContext context) {
+    fragmentOptions = context.getOptions();
+    final Iterator<OptionValue> optionList;
+    mergedOptions = sortOptions(fragmentOptions.iterator());
+  }
+  /*
+    Sort options according to name and the type to remove the redundant rows
+    for the same option based on the type
+   */
+  public Iterator<OptionValue> sortOptions(Iterator<OptionValue> options)
+  {
+    List<OptionValue> values = Lists.newArrayList(options);
+    List<OptionValue> optionValues = Lists.newArrayList();
+    List<OptionValue> optionValueList = Lists.newArrayList();
+    OptionValue value;
+    OptionType type;
+    OptionValue[] temp = new OptionValue[3];
+
+    for (int i = 0; i < values.size(); i++){
+      value = values.get(i);
+      final OptionValue def = SystemOptionManager.getValidator(value.name).getDefault();
+      if (value.equalsIgnoreType(def)) {
+        type = OptionType.DEFAULT;
+        optionValueList.add(OptionValue.createOption(value.kind,type,value.name,value.getValue().toString()));
+      } else {
+        type = value.type;
+        optionValueList.add(OptionValue.createOption(value.kind,type,value.name,value.getValue().toString()));
+      }
+    }
+    Collections.sort(optionValueList,  new Comparator<OptionValue>() {
+      @Override
+      public int compare(OptionValue v1, OptionValue v2) {
+        int nameCmp = v1.name.compareTo(v2.name);
+        if (nameCmp != 0) {
+          return nameCmp;
+        }
+        return v1.type.compareTo( v2.type);
+      }
+    });
+
+    for (int i = 0; i < optionValueList.size() ;i++ )
+    {
+      value = optionValueList.get(i);
+      type = value.type ;
+      switch (type) {
+        case DEFAULT:
+          temp[2] = value;
+          break;
+        case SYSTEM:
+          temp[1] = value;
+        case SESSION:
+          temp[0] = value;
+          break;
+      }
+      if(i == optionValueList.size() - 1 || (i < optionValueList.size()  && !value.getName().equals(optionValueList.get(i+1).getName()))) {
+        int j = 0;
+        while (temp[j] == null) {
+          j++;
+        }
+        optionValues.add(temp[j]);
+        Arrays.fill(temp,null);
+        }
+    }
+    return optionValues.iterator();
+  }
+
+  @Override
+  public boolean hasNext() {
+    return mergedOptions.hasNext();
+  }
+
+  @Override
+  public ExtendedOptionValueWrapper next() {
+    final OptionValue value = mergedOptions.next();
+    return new ExtendedOptionValueWrapper(value.name, value.kind, value.type,value.getValue().toString());
+  }
+
+  public enum Status {
+    BOOT, DEFAULT, CHANGED
+  }
+
+  /**
+   * Wrapper class for Extended Option Value
+   */
+  public static class ExtendedOptionValueWrapper {
+
+    public final String name;
+    public final Kind kind;
+    public final OptionType type;
+    public final String string_value;
```

Comment:
string_value --> value

Since this is a column name, might as well make it simple.

Also, can we include the data type? Each validator can return its type. For example, LongValidator can return either `Long.class` (kind of messy), or `OptionDataType.LONG` (which is a bit less messy). Then, convert the internal type to a nice display name, perhaps mapping to the DrillType. So, a long becomes `Bigint` or `BIGINT`.